### PR TITLE
YLP-460 Add footer login page

### PR DESCRIPTION
**Pull request by Nas84**

### CODE REVIEW

File: packages/yourloops/components/language-select.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2021, Diabeloop
```
Comment (yvon-dblg) [inline packages/yourloops/components/language-select.tsx:2]: I would have add a small text below the Copyright line, to describe the purpose of this file

File: packages/yourloops/pages/login/login.tsx [hunk 2]
```diff
@@ -192,6 +251,36 @@ function Login(props: RouteComponentProps): JSX.Element {
...
+              <Link href={diabeloopUrls.getTermsUrL(i18n.language)}>
+                {t(diabeloopUrls.Terms)}
+              </Link>
+            </Grid>
+            <Grid item xs={6} className={classes.leftLink}>
+              <Link href={diabeloopUrls.getIntendedUseUrL(i18n.language)}>
+                {t(diabeloopUrls.IntendedUse)}
+              </Link>
+            </Grid>
+          </Grid>
```
Comment (yvon-dblg) [inline packages/yourloops/pages/login/login.tsx:283]: I would have put this part in another functional component (an JSX function), to keep the "HTML" part more readable.
You may put this function in the same file, like I've done in https://github.com/mdblp/blip/blob/215edb4bb8c66bedde0735d74f29682d2565c517/packages/yourloops/pages/hcp/team-card.tsx#L115 for example.

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
Looks good.

- Need to merge or rebase from the base branch.

File: packages/yourloops/pages/login/login.tsx (hunk 2 above)
Comment (Nas84) [reply packages/yourloops/pages/login/login.tsx:283]: i think the same, i was going to do it later, but you motivated me to do it now 😉

File: packages/yourloops/lib/diabeloop-url.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+  constructor() {
+    this.rootPathName = `${config.ASSETS_URL}`;
+    this.termsUrl = `${this.rootPathName}terms.pdf`;
+    this.privacyPolicyUrl = `${this.rootPathName}data-privacy.pdf`;
+    this.intendedUseUrL = `${this.rootPathName}intended-use.pdf`;
+    this.supportUrL = "https://www.diabeloop.com";
+  }
+
+  get Support(): string {
```
Comment (yvon-dblg) [inline packages/yourloops/lib/diabeloop-url.ts:48]: I would have use lowercase for the getters in this class.
Not sure though.

Anyway, can you add
```
"lodash/prefer-constant": "off",
```
in the eslint project file ?

File: packages/yourloops/lib/diabeloop-url.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+import config from "./config";
+
+/**
+ * Class containing all URLs related to Diableloop
+ */
+class DiabeloopUrl {
```
Comment (yvon-dblg) [inline packages/yourloops/lib/diabeloop-url.ts:33]: I don't really understand the logic here.
Setting a private variables, without really using it.

Using a class instance, but all methods could be static.

Maybe you should make this class listen to i18next lang change in the constructor
If it is the case, update the URLs (private variables, like termsUrl)
and replace the functions likes getTermsUrL() with a getter ?

Also rootPathName bring nothing really interesting in that case.

File: packages/yourloops/pages/login/login-footer-link.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    },
+    selection: {
+      padding: theme.spacing(2),
+      textAlign: "center",
+      fontSize: "small",
+    },
+  };
+});
+
+function LoginFooterLink(): JSX.Element{
```
Comment (yvon-dblg) [inline packages/yourloops/pages/login/login-footer-link.tsx:38]:
```suggestion
function LoginFooterLink(): JSX.Element {

```
lint

File: packages/yourloops/pages/login/login-footer-link.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+import Link from "@material-ui/core/Link";
+
+import LanguageSelect from "../../components/language-select";
+import diabeloopUrls from "../../lib/diabeloop-url";
+import config from "../../lib/config";
+
+const loginStyle = makeStyles((theme: Theme) => {
+  return {
+    rightLink: {
+      padding: theme.spacing(0.5),
```
Comment (yvon-dblg) [inline packages/yourloops/pages/login/login-footer-link.tsx:16]:
```suggestion
      padding: theme.spacing(0.5), // eslint-disable-line no-magic-numbers

```

File: packages/yourloops/pages/login/login-footer-link.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+      textAlign: "start",
+      fontSize: "small",
+    },
+    centeredLink: {
+      padding: theme.spacing(0.5),
```
Comment (yvon-dblg) [inline packages/yourloops/pages/login/login-footer-link.tsx:21]:
```suggestion
      padding: theme.spacing(0.5), // eslint-disable-line no-magic-numbers

```

File: packages/yourloops/pages/login/login-footer-link.tsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+      textAlign: "center",
+      color: "#109182",
+    },
+    leftLink: {
+      padding: theme.spacing(0.5),
```
Comment (yvon-dblg) [inline packages/yourloops/pages/login/login-footer-link.tsx:26]:
```suggestion
      padding: theme.spacing(0.5), // eslint-disable-line no-magic-numbers

```

File: packages/yourloops/pages/login/login.tsx [hunk 9]
```diff
@@ -43,27 +43,50 @@ import Grid from "@material-ui/core/Grid";
...
-const loginStyle = makeStyles((/* theme: Theme */) => {
+const loginStyle = makeStyles((theme: Theme) => {
   return {
     mainContainer: { margin: "auto" },
+    root: { minHeight: "100vh" },
     loginButton: {
       marginLeft: "auto !important",
     },
+    rightLink: {
+      padding: theme.spacing(0.5),
```
Comment (yvon-dblg) [inline packages/yourloops/pages/login/login.tsx:63]:
```suggestion
      padding: theme.spacing(0.5), // eslint-disable-line no-magic-numbers
```

File: packages/yourloops/pages/login/login.tsx [hunk 10]
```diff
@@ -43,27 +43,50 @@ import Grid from "@material-ui/core/Grid";
...
+      textAlign: "start",
+      fontSize: "small",
+    },
+    centeredLink: {
+      padding: theme.spacing(0.5),
```
Comment (yvon-dblg) [inline packages/yourloops/pages/login/login.tsx:68]:
```suggestion
      padding: theme.spacing(0.5), // eslint-disable-line no-magic-numbers
```

File: packages/yourloops/pages/login/login.tsx [hunk 11]
```diff
@@ -43,27 +43,50 @@ import Grid from "@material-ui/core/Grid";
...
+      textAlign: "center",
+      color: "#109182",
+    },
+    leftLink: {
+      padding: theme.spacing(0.5),
```
Comment (yvon-dblg) [inline packages/yourloops/pages/login/login.tsx:73]:
```suggestion
      padding: theme.spacing(0.5), // eslint-disable-line no-magic-numbers
```

File: packages/yourloops/pages/login/login.tsx [hunk 12]
```diff
@@ -115,7 +142,9 @@ function Login(props: RouteComponentProps): JSX.Element {
     } catch (reason: unknown) {
       log.error(reason);
       setValidateError(true);
-      const message = _.isError(reason) ? reason.message : new String(reason).toString();
+      const message = _.isError(reason)
```
Comment (yvon-dblg) [inline packages/yourloops/pages/login/login.tsx:145]: I have add an utility function `errorTextFromException()` in `packages/yourloops/lib/utils.ts` if you want.

File: packages/yourloops/pages/login/login.tsx [hunk 13]
```diff
@@ -43,27 +43,50 @@ import Grid from "@material-ui/core/Grid";
...
+    },
   };
 });
 
 /**
  * Login page
  */
 function Login(props: RouteComponentProps): JSX.Element {
-  const { t } = useTranslation("yourloops");
+  const { t } = useTranslation();
```
Comment (yvon-dblg) [inline packages/yourloops/pages/login/login.tsx:89]: I would have prefer we use the new translation file.
But it need to be discuss with marine maybe.

File: packages/yourloops/pages/login/login.tsx [hunk 14]
```diff
@@ -134,15 +163,22 @@ function Login(props: RouteComponentProps): JSX.Element {
 
   return (
```
Comment (yvon-dblg) [inline packages/yourloops/pages/login/login.tsx:164]: Commented functions above can be removed ?

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
Can you reformat the `packages/yourloops/pages/login/login.tsx` code using the new rules (set in .vscode/settings.json) ?

File: packages/yourloops/pages/login/login.tsx (hunk 14 above)
Comment (Nas84) [reply packages/yourloops/pages/login/login.tsx:164]: i am going to change them soon in another PR

File: packages/yourloops/lib/diabeloop-url.ts (hunk 4 above)
Comment (Nas84) [reply packages/yourloops/lib/diabeloop-url.ts:33]: me too i am not satisfied with this class/file. but we agree to do some imperfect code which works and refactor.

File: packages/yourloops/lib/diabeloop-url.ts (hunk 3 above)
Comment (Nas84) [reply packages/yourloops/lib/diabeloop-url.ts:48]: sure

**Review by yvon-dblg — APPROVED** [review]
LGTM

**Pull request merged**